numpy/core/src/umath/matmul.c.src

```diff
@@ -0,0 +1,384 @@
+/* -*- c -*- */
+
+#define _UMATHMODULE
+#define _MULTIARRAYMODULE
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+
+#include "Python.h"
+
+#include "npy_config.h"
+#include "numpy/npy_common.h"
+#include "numpy/arrayobject.h"
+#include "numpy/ufuncobject.h"
+#include "numpy/npy_math.h"
+#include "numpy/halffloat.h"
+#include "lowlevel_strided_loops.h"
+
+#include "npy_pycompat.h"
+
+#include "npy_cblas.h"
+#include "arraytypes.h" /* For TYPE_dot functions */
+#include <assert.h>
+
+/*
+ *****************************************************************************
+ **                            BASICS                                       **
+ *****************************************************************************
+ */
+
+#if defined(HAVE_CBLAS)
+static const double oneD[2] = {1.0, 0.0}, zeroD[2] = {0.0, 0.0};
+static const float oneF[2] = {1.0, 0.0}, zeroF[2] = {0.0, 0.0};
```

Comment:
You can use `npy_cdouble` and `npy_cfloat` here, which would be a little clearer. The memory layouts are identical, so cblas won't know the difference.

Reply:
These are passed to `cblas*gemm` which requires `void*`, so if I change to `npy_cdouble` I have to cast (from `struct` to `void*`) where as it is the cast is automatic. I would prefer to leave this as it is.

Reply:
Yes, you do that with `&zeroD`

Reply:
done